File: c++/58_Length_Last_Word.cpp
#include <sstream>
#include <vector>
#include <string>
#include <iostream>

using namespace std;
// run time beats 3.8 % memory 5.1 %
//  very slow solution, but it works
class Solution {
public:
    int lengthOfLastWord(string s) {
    
    int len = 0;
    vector<string> words;

    stringstream ss(s);
    string word;

    // split words
    while (ss >> word) {
        words.push_back(word);
    }

    // get length of last word
    if (!words.empty()) {
        len = words.back().size();
    }

    return len;

    }

};

File: Python/58_Length_Last_Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        length = 0
        memory = False

        for char in s:
            if char != ' ' :
                if not memory:
                    memory = True
                    length = 1
                else:
                    length += 1
            else:
                memory = False

        return length
